flytekit/clis/flyte_cli/main.py

```diff
@@ -1098,17 +1110,42 @@ def execute_launch_plan(project, domain, name, host, insecure, urn, principal, v
     """
     _welcome_message()
 
-    with _platform_config.URL.get_patcher(host), _platform_config.INSECURE.get_patcher(_tt(insecure)):
-        lp_id = _identifier.Identifier.from_python_std(urn)
-        lp = _launch_plan_common.SdkLaunchPlan.fetch(lp_id.project, lp_id.domain, lp_id.name, lp_id.version)
+    execution = _execute_launch_plan(project, domain, name, host, insecure, urn, principal, verbose, lp_args)
+    _click.secho("Launched execution: {}".format(_tt(execution.id)), fg='blue')
+    _click.echo("")
 
-        inputs = _construct_literal_map_from_parameter_map(lp.default_inputs, _parse_args_into_dict(lp_args))
-        # TODO: Implement notification overrides
-        # TODO: Implement label overrides
-        # TODO: Implement annotation overrides
-        execution = lp.launch_with_literals(project, domain, inputs, name=name)
-        _click.secho("Launched execution: {}".format(_tt(execution.id)), fg='blue')
-        _click.echo("")
+
+@_flyte_cli.command('execute-launch-plan-and-wait', cls=_FlyteSubCommand)
```

Comment:
I think this should just be a flag on the existing execute-lp command (`--wait`)